resources/aws_ec2.go

```diff
@@ -20,25 +20,46 @@ package resources
 import (
 	"context"
 	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"log"
+	"net/http"
 	"sync"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/aws/session"
+	cwe "github.com/aws/aws-sdk-go/service/cloudwatchevents"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/sns"
+	multierr "github.com/hashicorp/go-multierror"
 	errwrap "github.com/pkg/errors"
 )
 
 func init() {
 	RegisterResource("aws:ec2", func() Res { return &AwsEc2Res{} })
 }
 
-// AwsPrefix is a const which gets prepended onto the instance name.
-const AwsPrefix = "mgmt:"
+const (
+	// AwsPrefix is a const which gets prepended onto the instance name.
+	AwsPrefix = "mgmt:"
+	// waitTimeout is the duration of the timeout context in CheckApply
+	waitTimeout = 400
+	// topicName is the name of the sns topic created by makeSnsTopic
+	topicName = "mgmtconfigEC2events"
```

Comment:
Does this work when we have multiple aws resources running simultaneously or is there a topic collision? In any case, It would nice if we built up these constants from a logical mgmt prefix. Eg:

```
const (
    AwsPrefix = "_mgmt"
    SnsPrefix = AwsPrefix + ":sns"
)
```

Or whatever makes sense logically and aesthetically. Same for the rule below.

Reply:
I guess it makes sense to have a unique topic for each resource, ie. SnsPrefix + obj.prependName(), that way, we don't see events from other instances.